tests/integration/pilot/security/authn_permissive_test.go

```diff
@@ -119,22 +125,26 @@ func TestAuthnPermissive(t *testing.T) {
 	pilot := pilot2.NewOrFail(t, ctx, pilot2.Config{})
 	aps := apps.NewOrFail(ctx, t, apps.Config{Pilot: pilot})
 	a := aps.GetAppOrFail("a", t)
-
-	req := apps.ConstructDiscoveryRequest(a, "type.googleapis.com/envoy.api.v2.Listener")
-	resp, err := pilot.CallDiscovery(req)
-	if err != nil {
-		t.Errorf("failed to call discovery %v", err)
-	}
-	for _, r := range resp.Resources {
-		foo := &xdsapi.Listener{}
-		if err := proto.UnmarshalAny(&r, foo); err != nil {
-			t.Errorf("failed to unmarshal %v", err)
+	endTime := time.Now().Add(timeout)
+	for {
```

Comment:
Would you mind switching over to the retry library under pkg/test/util/retry?

Reply:
+1

Reply:
Thanks for the comments.  I have changed the logic to use WatchDiscovery instead to allow retry to get the response.